internal/engine/pod.go

```diff
@@ -103,17 +103,18 @@ func matchPodChangeToManifest(state *store.EngineState, action k8swatch.PodChang
 }
 
 // Checks the runtime state if we're already tracking this pod.
-// If not, create a new tracking object.
+// If not, AND if the pod matches the current deploy, create a new tracking object.
 // Returns a store.Pod that the caller can mutate, and true
 // if this is the first time we've seen this pod.
-func trackPod(ms *store.ManifestState, action k8swatch.PodChangeAction) (*store.Pod, bool) {
+func maybeTrackPod(ms *store.ManifestState, action k8swatch.PodChangeAction) (*store.Pod, bool) {
 	pod := action.Pod
 	podID := k8s.PodIDFromPod(pod)
 	startedAt := pod.CreationTimestamp.Time
 	status := k8swatch.PodStatusToString(*pod)
 	ns := k8s.NamespaceFromPod(pod)
 	hasSynclet := sidecar.PodSpecContainsSynclet(pod.Spec)
 	runtime := ms.GetOrCreateK8sRuntimeState()
+	currentDeploy := runtime.HasOKPodTemplateSpecHash(pod)
```

Comment:
nit: I like putting "is", "has", etc in bool names, e.g. `isCurrentDeploy` (on first skim I thought you were putting the hash in a var called "deploy", which would be weird!)